### PR TITLE
Document fielddata.yaml fail, ignore vim swp files

### PR DESCRIPTION
**Pull request by awick**
### Description
* Ignore test files that don't end with .yaml so vim swp files are ignored
* fielddata.yaml test files if opensearch cluster was stopped with docker kill instead of docker stop, document the issue
* upgrade remaining actions/checkout@v3 to v4 to stop warnings

### Issues Resolved
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]
It's a fine workaround, but maybe we should collect all `.yaml` files instead? Add a test?

**Comment by dblock** [comment]
@awick I needed to exclude `docker-compose.yml` in another PR, you can build on top of https://github.com/opensearch-project/opensearch-api-specification/pull/472/commits/140c0fd1579e131a3d89f69fa549cdd14866aeda.

**Comment by awick** [comment]
@dblock  I think this is good now, and I don't think that failure is me.

**Review by dblock — DISMISSED** [review]

File: tools/src/tester/TestRunner.ts [hunk 1]
```diff
@@ -69,6 +69,9 @@ export default class TestRunner {
     const path = file === '' ? folder : `${folder}/${file}`
     const next_prefix = prefix === '' ? file : `${prefix}/${file}`
     if (fs.statSync(path).isFile()) {
+      if (!path.endsWith('.yaml')) {
+        return []
+      }
```
Comment (dblock) [inline tools/src/tester/TestRunner.ts:74]: This is fine, but if you look below there's `EXCLUDED_FILES.includes()`, feels like this is where we should be checking the .yaml extension as well to keep things in one place?

Comment (awick) [reply tools/src/tester/TestRunner.ts:74]: Happy to move it, when I looked at it that seemed to be when it was reading files in a directory, so this will catch even the case when a file was used on the command line.

So this would not run:
npm run test:spec -- --opensearch-insecure --tests-path tests/cat/badfile

but that is an edge case I guess. Let me know which you prefer.

Comment (awick) [reply tools/src/tester/TestRunner.ts:74]: or maybe i should move the EXCLUDE? Or actually the current list is a file that won't run anymore anyway, lol.

Comment (dblock) [reply tools/src/tester/TestRunner.ts:74]: Right. So we definitely don't want both checks, pick your favorite implementation and update?

**Review by dblock — APPROVED** [review]
